src/codegen.cpp

```diff
@@ -6432,6 +6432,7 @@ extern "C" void jl_init_codegen(void)
 #endif
 
 #ifdef USE_POLLY
+    cl::ParseEnvironmentOptions("Julia", "JULIA_POLLY_ARGS");
```

Comment:
The point of the user being able to use separate env-args even if both are applied the same way still stands, because he can (although he is not forced to) keep his setting separately.
A solution scoping the application and filtering the settings would be preferable and I'm not conclusively convinced llvm::cl cannot do that. I just don't know how, for now.